client/landing/gutenboarding/hooks/use-on-site-creation.ts

```diff
@@ -79,11 +80,11 @@ export default function useOnSiteCreation() {
 		if ( newSite && ! isRedirecting ) {
 			setIsRedirecting( true );
 
-			if ( selectedPlan && selectedPlan?.getStoreSlug() !== PLAN_FREE ) {
+			if ( selectedPlan && ! selectedPlan?.isFree ) {
 				const planProduct = {
-					meta: selectedPlan.getTitle(),
-					product_id: selectedPlan.getProductId(),
-					product_slug: selectedPlan.getStoreSlug(),
+					meta: selectedPlan.title,
+					product_id: selectedPlan.productId,
+					product_slug: selectedPlan.pathSlug,
```

Comment:
I think we need to use store slugs here which are different than `pathSlug` https://github.com/Automattic/wp-calypso/blob/73e47e4854ccbe4a5308bc9dab69d59ab51be186/client/lib/plans/constants.js#L14-L25
```suggestion
					product_slug: selectedPlan.storeSlug,
```